### PR TITLE
Feature/Authentication

### PR DESCRIPTION
**Pull request by aliashkevich**
Added Passport + JWT + bcrypt.
PUT, POST, DELETE methods require authentication. 
Passwords encrypted in Users.json.

### CODE REVIEW

File: routes/auth/auth.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    if (err) res.status(500).json({flash: err});
+    if (!user) res.status(401).json({flash: info.message});
+    if (user) {
+      var data = {
+        id: user.id,
+        name: user.name,
+      };
+
+      const token = jwt.sign(data, 'your_jwt_secret');
+      const flash = 'Sucessful login!';
```
Comment (laura-mso) [inline routes/auth/auth.js]:
```suggestion
      const flash = 'Successfully logged in!';
```

File: routes/auth/passport.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  new LocalStrategy(
+    {
+      usernameField: 'email',
+      passwordField: 'password',
+      session: false,
+    },
+    function(email, password, cb) {
+      let user = users.find(user => user.email === email);
+      if (user === undefined)
+        cb(null, false, {message: 'Incorrect email or password.'});
```
Comment (laura-mso) [inline routes/auth/passport.js]:
```suggestion
        cb(null, false, {message: 'Incorrect email.'});
```

**Comment by Wils0nC00k** [comment]
Wow such a big job! Can't wait to see it up and running. Well done

**Comment by Wils0nC00k** [comment]
Amazing! Cant find a fault

**Pull request merged**